File: frontend/src/services/userService.ts
import api from './api';
import type { ApiResponse, PageResponse, User } from '@/types';

export const userService = {
  async searchUsers(query: string, page = 0, size = 20): Promise<PageResponse<User>> {
    const response = await api.get<ApiResponse<PageResponse<User>>>('/users/search', {
      params: { q: query, page, size },
    });
    return response.data.data;
  },

  async getUserById(id: string): Promise<User> {
    const response = await api.get<ApiResponse<User>>(`/users/${id}`);
    return response.data.data;
  },

  async updateProfile(data: { displayName?: string; avatarUrl?: string }): Promise<User> {
    const response = await api.put<ApiResponse<User>>('/users/me', data);
    return response.data.data;
  },
};

File: backend/src/main/java/com/chatweb/user/service/UserService.java
package com.chatweb.user.service;

import com.chatweb.common.dto.PageResponse;
import com.chatweb.user.dto.UpdateProfileRequest;
import com.chatweb.user.dto.UserProfileResponse;
import com.chatweb.user.dto.UserSummaryResponse;

import java.util.UUID;

public interface UserService {

    /**
     * Retrieves full profile information for the authenticated user.
     *
     * @param currentUserId the ID of the current authenticated user
     * @return UserProfileResponse containing full profile details
     */
    UserProfileResponse getCurrentUserProfile(UUID currentUserId);

    /**
     * Updates profile details (display name, avatar URL) for the authenticated user.
     *
     * @param currentUserId the ID of the current authenticated user
     * @param request the update profile request
     * @return UserProfileResponse containing updated profile details
     */
    UserProfileResponse updateProfile(UUID currentUserId, UpdateProfileRequest request);

    /**
     * Retrieves public summary information of a user by their ID.
     *
     * @param targetUserId the ID of the user to view
     * @return UserSummaryResponse containing public user information
     */
    UserSummaryResponse getUserById(UUID targetUserId);

    /**
     * Searches for users by username or display name with pagination, excluding the current user.
     *
     * @param currentUserId the ID of the current user to exclude from results
     * @param query the search keyword
     * @param page page number (0-indexed)
     * @param size page size
     * @return PageResponse containing list of UserSummaryResponse
     */
    PageResponse<UserSummaryResponse> searchUsers(UUID currentUserId, String query, int page, int size);
}
